examples/templates/clusterclass.yaml

```diff
@@ -0,0 +1,459 @@
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: Metal3ClusterTemplate
+metadata:
+  name: example-cluster-template
+spec:
+  template:
+    spec:
+      controlPlaneEndpoint:
+        host: 127.0.0.1
+        port: 6443
+      noCloudProvider: true
+---
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+kind: KubeadmControlPlaneTemplate
+metadata:
+  name: example-controlplane
+spec:
+  template:
+    spec:
+      kubeadmConfigSpec:
+        clusterConfiguration:
+          apiServer:
+            extraArgs:
+              cloud-provider: baremetal
```

Comment:
What is this cloud provider?

Reply:
TBH, I've copied it from the existing example: https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/examples/controlplane/controlplane.yaml#L28

Reply:
Not sure about what this does honestly. I think we can remove it. We don't use it in the tests at least: https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/test/e2e/data/infrastructure-metal3/bases/cluster/cluster-with-kcp.yaml

Reply:
OK, removed all references to `cloud-provider: baremetal`.